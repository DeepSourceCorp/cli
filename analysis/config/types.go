package config

import "github.com/deepsourcelabs/cli/analysis/lsp"

///////////////////////////////////////////////////////
// DSConfig is the struct for .deepsource.toml file //
/////////////////////////////////////////////////////

type DSConfig struct {
	Version         int           `mapstructure:"version,omitempty" json:"version" toml:"version"`
	ExcludePatterns []string      `mapstructure:"exclude_patterns,omitempty" json:"exclude_patterns,omitempty" toml:"exclude_patterns"`
	TestPatterns    []string      `mapstructure:"test_patterns,omitempty" json:"test_patterns,omitempty" toml:"test_patterns"`
	Analyzers       []Analyzer    `mapstructure:"analyzers,omitempty" json:"analyzers,omitempty" toml:"analyzers"`
	Transformers    []Transformer `mapstructure:"transformers,omitempty" json:"transformers,omitempty" toml:"transformers"`
}

type Analyzer struct {
	Name                string      `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	RuntimeVersion      string      `mapstructure:"runtime_version,omitempty" json:"runtime_version,omitempty" toml:"runtime_version,omitempty"`
	Enabled             bool        `mapstructure:"enabled,omitempty" json:"enabled" toml:"enabled"`
	DependencyFilePaths []string    `mapstructure:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `mapstructure:"meta,omitempty" json:"meta,omitempty" toml:"meta"`
	Thresholds          interface{} `mapstructure:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	Enabled bool   `mapstructure:"enabled,omitempty" json:"enabled,omitempty" toml:"enabled"`
}

///////////////////////////////////////
// LSP based Analysis Config Types  //
/////////////////////////////////////

type AnalysisConfig struct {
	Files           []lsp.TextDocumentItem `json:"files"`
	TestFiles       []lsp.TextDocumentItem `json:"test_files"`
	ExcludedFiles   []lsp.TextDocumentItem `json:"excluded_files"`
	ExcludePatterns []string               `json:"exclude_patterns"`
	TestPatterns    []string               `json:"test_patterns"`
	AnalyzerMeta    interface{}            `json:"analyzer_meta"`
}

//////////////////////////////////////
// LSP based Analysis Result Types  //
/////////////////////////////////////

type Namespace struct {
	Key   string  `json:"key"`
	Value float64 `json:"value"`
}

type Metric struct {
	MetricCode string      `json:"metric_code"`
	Namespaces []Namespace `json:"namespaces"`
}

type AnalysisResult struct {
	Issues   []lsp.Diagnostic `json:"issues"`
	Metrics  []Metric         `json:"metrics,omitempty"`
	IsPassed bool             `json:"is_passed"`
	Errors   []Error          `json:"errors"`
	// Errors    []lsp.Diagnostic `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}

////////////////////////////////////
// Default Analysis Result Types  //
///////////////////////////////////

type Error struct {
	HMessage string `json:"hmessage"`
	Level    int    `json:"level"`
}

type Coordinate struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}

type Position struct {
	Begin Coordinate `json:"begin"`
	End   Coordinate `json:"end"`
}

type Location struct {
	Path     string   `json:"path"`
	Position Position `json:"position"`
}

type Issue struct {
	Code     string   `json:"issue_code"`
	Title    string   `json:"issue_text"`
	Location Location `json:"location"`
}

type DefaultAnalysisResult struct {
	Issues    []Issue     `json:"issues"`
	Metrics   []Metric    `json:"metrics,omitempty"`
	IsPassed  bool        `json:"is_passed"`
	Errors    []Error     `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}
