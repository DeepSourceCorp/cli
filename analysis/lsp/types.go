package lsp

//////////////////////////////
// Document identity types  //
/////////////////////////////

type DocumentURI string

type TextDocumentItem struct {
	URI        DocumentURI `json:"uri"`
	LanguageID string      `json:"languageID,omitempty"`
	Version    int         `json:"version,omitempty"`
	Text       string      `json:"text,omitempty"`
}

///////////////////////////////
// Diagnostic related types //
/////////////////////////////

type DiagnosticSeverity int

const (
	Error       DiagnosticSeverity = 1
	Warning     DiagnosticSeverity = 2
	Information DiagnosticSeverity = 3
	Hint        DiagnosticSeverity = 4
)

type Position struct {
	Line      int `json:"line"`
	Character int `json:"character"`
}
type Range struct {
	Start Position `json:"start"`
	End   Position `json:"end"`
}

type Location struct {
	URI   string `json:"uri"`
	Range Range  `json:"range"`
}

type DiagnosticRelatedInformation struct {
	Location Location `json:"location"`
	Message  string   `json:"message"`
}

type Diagnostic struct {
	Range    Range              `json:"range"`
	Severity DiagnosticSeverity `json:"severity,omitempty"`
	Code     string             `json:"code,omitempty"`
	Source   string             `json:"source,omitempty"`
	Message  string             `json:"message"`

	/**
	 * An array of related diagnostic information, e.g. when symbol-names within
	 * a scope collide all definitions can be marked via this property.
	 * var a,b
	 * a := 2
	 * Issues in line 1 and 2 are related.
	 */
	RelatedInformation []DiagnosticRelatedInformation `json:"relatedInformation"`
}
