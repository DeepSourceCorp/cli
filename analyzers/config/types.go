package config

type Analysis struct {
	Command string `toml:"command" json:"command" validate:"required"`
}

// TODO: Remove the comment when we start support for Autofix.
// type Autofix struct {
//     Command string `toml:"command" json:"command" analyzertoml:"required"`
// }

type Build struct {
	Builder    string `toml:"builder" json:"builder" validate:"omitempty"`
	Dockerfile string `toml:"dockerfile" json:"dockerfile,omitempty" validate:"omitempty"`
	Script     string `toml:"script" json:"script" validate:"omitempty"`
}

type URL struct {
	Source        string `toml:"source" json:"source" validate:"omitempty,url"`
	Documentation string `toml:"documentation" json:"documentation,omitempty" validate:"omitempty,url"`
	BugTracker    string `toml:"bug_tracker" json:"bug_tracker" validate:"omitempty,url"`
}

type Test struct {
	Command string `toml:"command" json:"command" validate:"omitempty"`
}

// analyzer.toml type
type AnalyzerMetadata struct {
	// Analyzer specific data
	Name                 string            `toml:"name" json:"name" validate:"required"`
	Shortcode            string            `toml:"shortcode" json:"shortcode" validate:"required"` // New
	Description          string            `toml:"description" json:"description" validate:"required"`
	Category             string            `toml:"category" json:"category" validate:"required"`
	Tags                 []string          `toml:"tags" json:"tags,omitempty" validate:"omitempty"`
	URL                  URL               `toml:"urls" json:"urls,omitempty" validate:"omitempty"`
	EnvironmentVariables map[string]string `toml:"environment_variables" json:"environment_variables,omitempty" validate:"omitempty"`

	// Analyzer, Autofix and Transformer config
	Analysis Analysis `toml:"analysis" json:"analysis" validate:"required"`
	// Autofix Autofix `toml:"autofix" json:"autofix" validate:"omitempty"`

	// Build steps needed to build the analyzer
	Build Build `toml:"build" json:"build" validate:"omitempty"`

	// Test command for running the testing workflow
	Test Test `toml:"test" json:"test" validate:"omitempty"`
}

// Analyzer issue description
type AnalyzerIssue struct {
	Shortcode   string `validate:"omitempty"`
	Title       string `toml:"title" json:"title" validate:"required"`
	Description string `toml:"description" json:"description" validate:"required"`
	Category    string `toml:"category" json:"category" validate:"required"`
}
