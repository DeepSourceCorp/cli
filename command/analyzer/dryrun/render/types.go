package render

import (
	"html/template"
	"time"

	"github.com/deepsourcelabs/cli/types"
	"github.com/labstack/echo/v4"
)

type RunSummary struct {
	RunDuration       string // Time taken to complete analysis.
	TimeSinceRun      string // Time elapsed since the completion of the analysis run.
	AnalysisStartTime time.Time
	AnalysisEndTime   time.Time
}

type VCSInfo struct {
	Branch      string // VCS branch of the Analyzer.
	CommitSHA   string // The latest commit SHA of the Analyzer.
	VersionDiff string // The string specifying the status of Analyzer w.r.t previous version.
}

type OccurenceData struct {
	IssueMeta  types.AnalyzerIssue // Contains the data stored in issue TOMLs for the respective issue.
	Files      []string            // Files where this issue has been reported.
	FilesInfo  string              // The string containing the data of which files the issue has been reported in.
	Occurences []types.Issue       // The slice of occurences for a certain issue code.
}

type ResultData struct {
	UniqueIssuesCount     int                      // The unique issues count.
	TotalOccurences       int                      // Total issues reported by the Analyzer.
	SourcePath            string                   // The path where the source code to be analyzer is stored.
	IssuesOccurenceMap    map[string]OccurenceData // The map of issue code to its occurences data.
	IssueCategoryCountMap map[string]int           // The map of issue category to the count of the issues of that category.
	AnalysisResult        types.AnalysisResult     // The analysis result post running processors.
	MetricsMap            map[string]float64       // The map of metric names to their values.
	RenderedSourceCode    []template.HTML          // The slice containing the source code snippets for each occurence.
}

type ResultRenderOpts struct {
	EchoServer           *echo.Echo         // The Echo server instance to run the renderer server.
	Template             *template.Template // The go template field so that it can be accessible in `route.go` as well.
	PageTitle            string             // The title of the HTML page.
	AnalyzerShortcode    string             // The shortcode of the Analyzer.
	VCSInfo              VCSInfo            // The VCS information of the Analyzer.
	Summary              RunSummary         // The run summary.
	AnalysisResultData   ResultData         // The analysis result data.
	SelectedIssueCode    string             // The field used to recognize which issue code the user has clicked on to check its occurences.
	SelectedCategory     string             // The field used to recognize which category the user has clicked to filter the issues based on it.
	IssueCategoryNameMap map[string]string  // The map used to route category names to their codes. Eg: `Documentation`->`doc`.
	MetricNameMap        map[string]string  // The map of metrics shortcodes with their names.
}
