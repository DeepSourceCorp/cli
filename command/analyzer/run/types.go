package run

type AnalysisConfig struct {
	Files           []string    `json:"files"`
	ExcludePatterns []string    `json:"exclude_patterns"`
	ExcludeFiles    []string    `json:"exclude_files"`
	TestFiles       []string    `json:"test_files"`
	TestPatterns    []string    `json:"test_patterns"`
	AnalyzerMeta    interface{} `json:"analyzer_meta"`
}
