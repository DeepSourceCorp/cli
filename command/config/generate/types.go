package generate

// DSConfig is the struct for .deepsource.toml file
type Analyzer struct {
	Name                string      `toml:"name,omitempty" json:"name,omitempty"`
	RuntimeVersion      string      `toml:"runtime_version,omitempty" json:"runtime_version,omitempty"`
	Enabled             bool        `toml:"enabled" json:"enabled"`
	DependencyFilePaths []string    `toml:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `toml:"meta,omitempty" json:"meta,omitempty"`
	Thresholds          interface{} `toml:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `toml:"name" json:"name"`
	Enabled bool   `toml:"enabled" json:"enabled"`
}

type DSConfig struct {
	Version         int           `toml:"version" json:"version"`
	ExcludePatterns []string      `toml:"exclude_patterns" json:"exclude_patterns,omitempty"`
	TestPatterns    []string      `toml:"test_patterns" json:"test_patterns,omitempty"`
	Analyzers       []Analyzer    `toml:"analyzers,omitempty" json:"analyzers,omitempty"`
	Transformers    []Transformer `toml:"transformers,omitempty" json:"transformers,omitempty"`
}
