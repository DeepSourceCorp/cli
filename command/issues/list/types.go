package list

// custom types for JSON marshaling

type IssueJSON struct {
	Analyzer       string       `json:"analyzer"`
	IssueCode      string       `json:"issue_code"`
	IssueTitle     string       `json:"issue_title"`
	OccurenceTitle string       `json:"occurence_title"`
	IssueCategory  string       `json:"issue_category"`
	Location       LocationJSON `json:"location"`
}

type LocationJSON struct {
	Path     string       `json:"path"`     // The filepath where the issue is reported
	Position PositionJSON `json:"position"` // The position info where the issue is raised
}

type PositionJSON struct {
	Begin LineColumn `json:"begin"` // The line where the code covered under the issue starts
	End   LineColumn `json:"end"`   // The line where the code covered under the issue starts
}

type LineColumn struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}
