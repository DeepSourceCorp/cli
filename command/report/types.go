package report

// ReportQueryInput is the schema for variables of artifacts
// report GraphQL query
type ReportQueryInput struct {
	AccessToken       string      `json:"accessToken"`
	CommitOID         string      `json:"commitOid"`
	ReporterName      string      `json:"reporter"`
	ReporterVersion   string      `json:"reporterVersion"`
	Key               string      `json:"key"`
	Data              string      `json:"data"`
	AnalyzerShortcode string      `json:"analyzer"`
	Metadata          interface{} `json:"metadata,omitempty"`
}

// ReportQueryInput is the structure of artifacts report
// GraphQL query
type ReportQuery struct {
	Query     string `json:"query"`
	Variables struct {
		Input ReportQueryInput `json:"input"`
	} `json:"variables"`
}

// QueryResponse is the response returned by artifacts report
// GraphQL query
type QueryResponse struct {
	Data struct {
		CreateArtifact struct {
			Error string `json:"error"`
			Ok    bool   `json:"ok"`
		} `json:"createArtifact"`
	} `json:"data"`
}
