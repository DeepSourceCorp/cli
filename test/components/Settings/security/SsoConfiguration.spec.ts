import { render } from '@testing-library/vue'
import { RouterLinkStub, shallowMount } from '@vue/test-utils'

import SsoConfiguration from '~/components/Settings/Security/SsoConfiguration.vue'

import { SSO_SETUP_STATES } from '~/types/cloudSso'
import { IdentityProvider, Maybe, VerifiableDomain } from '~/types/types'

import { IDENTIY_PROVIDER_MOCK, VERIFIABLE_DOMAIN_MOCK } from './__mocks__/verifiableDomain.mock'

interface SsoConfigurationT extends Vue {
  verifyDomain: () => void
}

describe('[[ SsoConfiguration ]]', () => {
  const baseProps = {
    verifiableDomain: null as Maybe<VerifiableDomain>,
    identityProvider: null as Maybe<IdentityProvider>,
    domainName: 'asdf',
    verifyingDomain: false,
    isInProgress: false,
    canSetupSso: true
  }

  const stubs = { ZAlert: true, ZButton: true, ZIcon: true, ZInput: true, NuxtLink: RouterLinkStub }

  test('renders correctly with all SSO setup states', () => {
    const ssoStatesProps: Record<SSO_SETUP_STATES, typeof baseProps> = {
      [SSO_SETUP_STATES.VERIFICATION_IN_PROGRESS]: { ...baseProps, isInProgress: true },
      [SSO_SETUP_STATES.NEED_TO_UPGRADE]: { ...baseProps, canSetupSso: false },
      [SSO_SETUP_STATES.NEEDS_INFORMATION]: { ...baseProps, verifiableDomain: null },
      [SSO_SETUP_STATES.VERIFICATION_COMPLETE]: {
        ...baseProps,
        verifiableDomain: VERIFIABLE_DOMAIN_MOCK
      }
    }
    Object.values(ssoStatesProps).forEach((props) => {
      const { html } = render(SsoConfiguration, {
        props,
        stubs
      })

      expect(html()).toMatchSnapshot()
    })
  })

  test('renders configured IDP data correctly', () => {
    const { html } = render(SsoConfiguration, {
      props: { ...baseProps, identityProvider: IDENTIY_PROVIDER_MOCK },
      stubs
    })

    expect(html()).toMatchSnapshot()
  })

  describe('[[verifyDomain]] method', () => {
    const mocks = { $emit: jest.fn(), $toast: { danger: jest.fn() } }

    const getInstance = (props: Partial<typeof baseProps>) => {
      const { vm } = shallowMount(SsoConfiguration, {
        propsData: { ...baseProps, ...props },
        mocks,
        stubs
      })

      return vm as SsoConfigurationT
    }

    test('emits event successfully', () => {
      const vm = getInstance({})
      vm.verifyDomain()
      expect(mocks.$emit).toBeCalledWith('verify-domain')
    })

    test('throws error successfully', () => {
      const vm = getInstance({ domainName: '' })
      vm.verifyDomain()
      expect(mocks.$toast.danger).toBeCalled()
    })
  })
})
