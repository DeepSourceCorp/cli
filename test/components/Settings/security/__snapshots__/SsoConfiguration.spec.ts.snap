// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ SsoConfiguration ]] renders configured IDP data correctly 1`] = `
"<div class=\\"\\">
  <div class=\\"space-y-5\\">
    <h2 class=\\"text-sm leading-5 text-vanilla-100\\">SAML Single Sign-On</h2>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Domain name</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        When configured, users whose email address ends in the specified domain would be allowed
        to sign in using your SSO provider.
        <a href=\\"https://docs.deepsource.com/docs/sso\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"inline-flex items-center gap-x-0.5 text-vanilla-100\\"><span>Learn more</span>
          <zicon-stub icon=\\"arrow-right\\" color=\\"current\\" size=\\"x-small\\"></zicon-stub></a></p>
      <div class=\\"mt-3 flex w-full items-stretch divide-x divide-ink-200 rounded-sm border border-ink-200 bg-ink-400\\">
        <zinput-stub padding=\\"p-0 px-3\\" placeholder=\\"yourdomain.com\\" value=\\"asdf\\" class=\\"flex-grow text-xs leading-5\\"></zinput-stub> <button class=\\"flex flex-shrink-0 items-center gap-x-1 bg-ink-300 p-3 text-xs leading-3 text-vanilla-100 hover:bg-ink-200\\">
          <zicon-stub icon=\\"link-2\\" color=\\"vanilla-100\\" size=\\"x-small\\" class=\\"\\"></zicon-stub> <span>Request verification</span>
        </button>
      </div>
      <!---->
    </div>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <div class=\\"float-right rounded-sm border border-ink-50 bg-ink-200 opacity-60\\">
        <zbutton-stub disabled=\\"true\\" button-type=\\"ghost\\" color=\\"vanilla-100\\" icon=\\"sliders\\" icon-size=\\"x-small\\" label=\\"Configure\\" size=\\"small\\"></zbutton-stub>
      </div>
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Identity provider</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        Allow team members to login using an Identity Provider.
      </p>
      <zalert-stub type=\\"info\\" class=\\"mt-3\\">
        <div class=\\"flex items-center gap-x-2\\">
          <zicon-stub icon=\\"solid-alert-circle\\" color=\\"robin-400\\"></zicon-stub> <span>You need to first verify your domain to configure SAML Single Sign-On.</span>
        </div>
      </zalert-stub>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`[[ SsoConfiguration ]] renders correctly with all SSO setup states 1`] = `
"<div class=\\"\\">
  <div class=\\"space-y-5\\">
    <h2 class=\\"text-sm leading-5 text-vanilla-100\\">SAML Single Sign-On</h2>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Domain name</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        When configured, users whose email address ends in the specified domain would be allowed
        to sign in using your SSO provider.
        <a href=\\"https://docs.deepsource.com/docs/sso\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"inline-flex items-center gap-x-0.5 text-vanilla-100\\"><span>Learn more</span>
          <zicon-stub icon=\\"arrow-right\\" color=\\"current\\" size=\\"x-small\\"></zicon-stub></a></p>
      <div class=\\"mt-3 flex w-full items-stretch divide-x divide-ink-200 rounded-sm border border-ink-200 bg-ink-400\\">
        <zinput-stub padding=\\"p-0 px-3\\" placeholder=\\"yourdomain.com\\" value=\\"asdf\\" class=\\"flex-grow text-xs leading-5\\"></zinput-stub> <button class=\\"flex flex-shrink-0 items-center gap-x-1 bg-ink-300 p-3 text-xs leading-3 text-vanilla-100 hover:bg-ink-200\\">
          <zicon-stub icon=\\"link-2\\" color=\\"vanilla-100\\" size=\\"x-small\\" class=\\"\\"></zicon-stub> <span>Request verification</span>
        </button>
      </div>
      <!---->
    </div>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <div class=\\"float-right rounded-sm border border-ink-50 bg-ink-200 opacity-60\\">
        <zbutton-stub disabled=\\"true\\" button-type=\\"ghost\\" color=\\"vanilla-100\\" icon=\\"sliders\\" icon-size=\\"x-small\\" label=\\"Configure\\" size=\\"small\\"></zbutton-stub>
      </div>
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Identity provider</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        Allow team members to login using an Identity Provider.
      </p>
      <zalert-stub type=\\"info\\" class=\\"mt-3\\">
        <div class=\\"flex items-center gap-x-2\\">
          <zicon-stub icon=\\"solid-alert-circle\\" color=\\"robin-400\\"></zicon-stub> <span>You need to first verify your domain to configure SAML Single Sign-On.</span>
        </div>
      </zalert-stub>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`[[ SsoConfiguration ]] renders correctly with all SSO setup states 2`] = `
"<div class=\\"cursor-not-allowed\\">
  <div class=\\"space-y-5 pointer-events-none opacity-60\\">
    <h2 class=\\"text-sm leading-5 text-vanilla-100\\">SAML Single Sign-On</h2>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Domain name</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        When configured, users whose email address ends in the specified domain would be allowed
        to sign in using your SSO provider.
        <a href=\\"https://docs.deepsource.com/docs/sso\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"inline-flex items-center gap-x-0.5 text-vanilla-100\\"><span>Learn more</span>
          <zicon-stub icon=\\"arrow-right\\" color=\\"current\\" size=\\"x-small\\"></zicon-stub></a></p>
      <div class=\\"mt-3 flex w-full items-stretch divide-x divide-ink-200 rounded-sm border border-ink-200 bg-ink-400\\">
        <zinput-stub padding=\\"p-0 px-3\\" placeholder=\\"yourdomain.com\\" value=\\"asdf\\" class=\\"flex-grow text-xs leading-5\\"></zinput-stub> <button class=\\"flex flex-shrink-0 items-center gap-x-1 bg-ink-300 p-3 text-xs leading-3 text-vanilla-100 hover:bg-ink-200\\">
          <zicon-stub icon=\\"link-2\\" color=\\"vanilla-100\\" size=\\"x-small\\" class=\\"\\"></zicon-stub> <span>Request verification</span>
        </button>
      </div>
      <!---->
    </div>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <div class=\\"float-right rounded-sm border border-ink-50 bg-ink-200 opacity-60\\">
        <zbutton-stub disabled=\\"true\\" button-type=\\"ghost\\" color=\\"vanilla-100\\" icon=\\"sliders\\" icon-size=\\"x-small\\" label=\\"Configure\\" size=\\"small\\"></zbutton-stub>
      </div>
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Identity provider</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        Allow team members to login using an Identity Provider.
      </p>
      <zalert-stub type=\\"info\\" class=\\"mt-3\\">
        <div class=\\"flex items-center gap-x-2\\">
          <zicon-stub icon=\\"solid-alert-circle\\" color=\\"robin-400\\"></zicon-stub> <span>You need to first verify your domain to configure SAML Single Sign-On.</span>
        </div>
      </zalert-stub>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`[[ SsoConfiguration ]] renders correctly with all SSO setup states 3`] = `
"<div class=\\"\\">
  <div class=\\"space-y-5\\">
    <h2 class=\\"text-sm leading-5 text-vanilla-100\\">SAML Single Sign-On</h2>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Domain name</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        When configured, users whose email address ends in the specified domain would be allowed
        to sign in using your SSO provider.
        <a href=\\"https://docs.deepsource.com/docs/sso\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"inline-flex items-center gap-x-0.5 text-vanilla-100\\"><span>Learn more</span>
          <zicon-stub icon=\\"arrow-right\\" color=\\"current\\" size=\\"x-small\\"></zicon-stub></a></p>
      <div class=\\"mt-3 flex w-full items-stretch divide-x divide-ink-200 rounded-sm border border-ink-200 bg-ink-400\\">
        <zinput-stub padding=\\"p-0 px-3\\" placeholder=\\"yourdomain.com\\" value=\\"asdf\\" class=\\"flex-grow text-xs leading-5\\"></zinput-stub> <button class=\\"flex flex-shrink-0 items-center gap-x-1 bg-ink-300 p-3 text-xs leading-3 text-vanilla-100 hover:bg-ink-200\\">
          <zicon-stub icon=\\"link-2\\" color=\\"vanilla-100\\" size=\\"x-small\\" class=\\"\\"></zicon-stub> <span>Request verification</span>
        </button>
      </div>
      <!---->
    </div>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <div class=\\"float-right rounded-sm border border-ink-50 bg-ink-200 opacity-60\\">
        <zbutton-stub disabled=\\"true\\" button-type=\\"ghost\\" color=\\"vanilla-100\\" icon=\\"sliders\\" icon-size=\\"x-small\\" label=\\"Configure\\" size=\\"small\\"></zbutton-stub>
      </div>
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Identity provider</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        Allow team members to login using an Identity Provider.
      </p>
      <zalert-stub type=\\"info\\" class=\\"mt-3\\">
        <div class=\\"flex items-center gap-x-2\\">
          <zicon-stub icon=\\"solid-alert-circle\\" color=\\"robin-400\\"></zicon-stub> <span>You need to first verify your domain to configure SAML Single Sign-On.</span>
        </div>
      </zalert-stub>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`[[ SsoConfiguration ]] renders correctly with all SSO setup states 4`] = `
"<div class=\\"\\">
  <div class=\\"space-y-5\\">
    <h2 class=\\"text-sm leading-5 text-vanilla-100\\">SAML Single Sign-On</h2>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Domain name</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        When configured, users whose email address ends in the specified domain would be allowed
        to sign in using your SSO provider.
        <a href=\\"https://docs.deepsource.com/docs/sso\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"inline-flex items-center gap-x-0.5 text-vanilla-100\\"><span>Learn more</span>
          <zicon-stub icon=\\"arrow-right\\" color=\\"current\\" size=\\"x-small\\"></zicon-stub></a></p>
      <div class=\\"juniper-highlight-card mt-3\\">
        <div class=\\"flex items-center gap-x-2\\">
          <div class=\\"h-1.5 w-1.5 rounded-full bg-juniper-500\\"></div> <span class=\\"text-sm leading-4 text-juniper-300\\">https://deepsource.io is verified.</span>
        </div>
      </div>
      <hr class=\\"-mx-4 mb-3 mt-4 border-ink-200\\">
      <div class=\\"-mb-1 flex flex-wrap items-center justify-between\\">
        <p class=\\"text-xs leading-4 text-vanilla-400\\">
          Please contact us to update your team's domain.
        </p> <a class=\\"flex items-center gap-x-1 rounded-sm border border-ink-50 bg-ink-200 px-2.5 py-1.5 text-xs leading-none\\">
          <zicon-stub icon=\\"mail\\" size=\\"x-small\\"></zicon-stub> <span>Contact support</span>
        </a>
      </div>
    </div>
    <div class=\\"rounded-md border border-ink-200 bg-ink-300 p-4\\">
      <div class=\\"float-right rounded-sm border border-ink-50 bg-ink-200\\">
        <zbutton-stub button-type=\\"ghost\\" color=\\"vanilla-100\\" icon=\\"sliders\\" icon-size=\\"x-small\\" label=\\"Configure\\" size=\\"small\\"></zbutton-stub>
      </div>
      <p class=\\"text-sm leading-4 tracking-wide text-vanilla-300\\">Identity provider</p>
      <p class=\\"mt-1.5 text-xs leading-5 text-vanilla-400\\">
        Allow team members to login using an Identity Provider.
      </p>
      <!---->
      <!---->
    </div>
  </div>
</div>"
`;
